crossdock/services/tracehandler.go

```diff
@@ -104,6 +141,78 @@ func (h *TraceHandler) runTest(service string, request *traceRequest, tFunc test
 	return nil
 }
 
+func (h *TraceHandler) adaptiveSamplingTest(service string, request *traceRequest) ([]*ui.Trace, error) {
+	stop := make(chan struct{})
+	go h.createTracesLoop(service, *request, stop)
+	defer close(stop)
+
+	var rate float64
+	var err error
+	for i := 0; i < 20; i++ {
+		// Keep checking to see if the sampling rate has been calculated
+		h.logger.Info(fmt.Sprintf("Waiting for adaptive sampling probabilities, iteration %d out of 20", i+1))
+		rate, err = h.agent.GetSamplingRate(service, request.Operation)
```

Comment:
why is the error check gets mixed with biz functionality? what if you always exit on error? I assume the agent is up by the time this starts